Client/Frontend/Toolbar+URLBar/TabLocationView.swift

```diff
@@ -290,7 +289,13 @@ class TabLocationView: UIView, FeatureFlaggable {
     @objc
     func didPressShoppingButton(_ button: UIButton) {
         button.isSelected = true
-        delegate?.tabLocationViewDidTapShopping(self, button: button)
+        TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .shoppingButton)
+
+        guard let fakespotState = store.state.screenState(FakespotState.self, for: .fakespot) else {
```

Comment:
I think the usage of the fakespotState directly from the store is justified here I just wanted to double-check with @OrlaM

Reply:
Update LGTM, I'll let @yoanarios give it the final sign off.